test/e2e/conformance/webhook_test.go

```diff
@@ -0,0 +1,292 @@
+	logicalClusters := []logicalcluster.LogicalCluster{
+		framework.NewWorkspaceFixture(t, server, organization, "Universal"),
+		framework.NewWorkspaceFixture(t, server, organization, "Universal"),
+	}
+
+	kubeClusterClient, err := kubernetes.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct client for server")
+	cowbyClients, err := client.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct cowboy client for server")
+	apiExtensionsClients, err := apiextensionsclient.NewClusterForConfig(cfg)
+	require.NoError(t, err, "failed to construct apiextensions client for server")
+
+	t.Logf("Install the Cowboy resources into logical clusters")
+	for _, logicalCluster := range logicalClusters {
+		t.Logf("Bootstrapping ClusterWorkspace CRDs in logical cluster %s", logicalCluster)
+		crdClient := apiExtensionsClients.Cluster(logicalCluster).ApiextensionsV1().CustomResourceDefinitions()
+		wildwest.Create(t, crdClient, metav1.GroupResource{Group: "wildwest.dev", Resource: "cowboys"})
+	}
+
+	t.Logf("Installing webhook into the first workspace")
+	sideEffect := admissionregistrationv1.SideEffectClassNone
+	url := testWebhook.GetURL()
+	webhook := &admissionregistrationv1.MutatingWebhookConfiguration{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{Name: "test-webhook"},
+		Webhooks: []admissionregistrationv1.MutatingWebhook{{
+			Name: "test-webhook.cowboy.io",
+			ClientConfig: admissionregistrationv1.WebhookClientConfig{
+				URL:      &url,
+				CABundle: cfg.CAData,
+			},
+			Rules: []admissionregistrationv1.RuleWithOperations{{
+				Operations: []admissionregistrationv1.OperationType{
+					admissionregistrationv1.Create,
+				},
+				Rule: admissionregistrationv1.Rule{
+					APIGroups:   []string{"wildwest.dev"},
+					APIVersions: []string{"v1alpha1"},
+					Resources:   []string{"cowboys"},
+				},
+			}},
+			SideEffects:             &sideEffect,
+			AdmissionReviewVersions: []string{"v1"},
+		}},
+	}
+	_, err = kubeClusterClient.Cluster(logicalClusters[0]).AdmissionregistrationV1().MutatingWebhookConfigurations().Create(ctx, webhook, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to add validating webhook configurations")
+
+	cowboy := v1alpha1.Cowboy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "testing",
+		},
+		Spec: v1alpha1.CowboySpec{},
+	}
+
+	// Avoid race condition here by making sure that CRD is served after installing the types into logical clusters
+	t.Logf("Creating cowboy resource in second logical cluster")
+	require.Eventually(t, func() bool {
+		_, err = cowbyClients.Cluster(logicalClusters[1]).WildwestV1alpha1().Cowboys("default").Create(ctx, &cowboy, metav1.CreateOptions{})
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return false
+		}
+		return true
+
+	}, wait.ForeverTestTimeout, 100*time.Millisecond)
+	require.Equal(t, 0, testWebhook.Calls(), "expected that the webhook is not called for logical cluster where webhook is not installed")
```

Comment:
this is not helpful. You don't know that the webhook informer in the apiserver is actually up-to-date and would call the webhook if there were a bug.

Reply:
What you can do:
1. create webhook
2. ```
    eventually {
       create in first
       return calls == 1
    }
    create in second
    require calls == 1
   ```

Reply:
Done